postgres/up/05-create_filesystem_table.sql

```diff
@@ -61,6 +62,24 @@ BEGIN
   RETURN newEntityID;
 END $$;
 
+-- TODO: Add a delete frontend function
+DROP FUNCTION IF EXISTS new_frontend;
+CREATE OR REPLACE FUNCTION new_frontend (frontendIDP uuid, logicalNameP VARCHAR, URLP VARCHAR) RETURNS uuid
+LANGUAGE plpgsql
+AS $$
+DECLARE
+  frontendID frontend.ID%type;
+BEGIN
+  INSERT INTO frontend VALUES (frontendIDP, logicalNameP, URLP)
+    RETURNING ID INTO frontendID;
+       
+  -- TODO: Temporarily setting the OwnedBy field. Going to be deprecated soon.
+  INSERT INTO filesystem (EntityID, LogicalName, IsDocument, Parent, OwnedBy)
```

Comment:
and then create a new file tree where the ID of the root is the frontendID, basically you overload the value you generate when creating a frontend to act as both the ID for the FE table and the ID for the root entity for that table

Reply:
IMO u shouldn't do this, you should maintain a separate column within the FE table that references that root, additionally we should have a foreign key constraint somewhere that enforces that